AsyncDisplayKit/ASTextNode+Subclasses.h

```diff
@@ -0,0 +1,27 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <AsyncDisplayKit/ASTextNode.h>
+
+/**
+ * The subclass header _ASTextNode+Subclasses_ defines methods to be
+ * overridden by custom nodes that subclass ASTextNode.
+ *
+ * These methods should never be called directly by other classes.
+ */
+
+@interface ASTextNode (Subclasses)
+
+/**
+ @abstract Calculates a size that will fit all of the text in this text node.
+ @param constrainedSize The maximum available space to render into
+ @discussion The size returned from this method is used to create an ASLayout of the appropriate size.
+ */
+- (CGSize)renderSizeThatFits:(CGSize)constrainedSize;
```

Comment:
I think you are right. I'm not sure I'm really gaining anything by having ASTextNode implement `calculateLayoutThatFits:` instead of `calculateSizeThatFits:` if I've overriding `calculateLayoutThatFits:` anyway. Thinking about it now, what I'm trying to do with these subclasses is probably something that does not need to live in ASDK, but in subclasses in my own project. Let me play around with that and see if I can get it to work, and perhaps we can abandon this PR.